### ISSUE TITLE
imagenet/multiedge_inference_bench: save_weights() stores PyTorch tensors instead of numpy arrays in deit.py

### ISSUE THREAD

**Issue by Nityahiray** [kubeedge/ianvs#485]
**What happened**:
In deit.py, the save_weights() function in 
DeiTModelShard stores PyTorch tensor objects 
directly into a dictionary and then calls 
np.savez() to save them. But np.savez() only 
supports numpy arrays, not PyTorch tensors. 
This crashes with a TypeError when anyone tries 
to save model weights.

**What you expected to happen**:
The model weights should be saved successfully 
as a .npz file without any errors.

**How to reproduce it**:
1. Go to examples/imagenet/multiedge_inference_bench
2. Run DeiTModelShard.save_weights()
3. The program crashes with TypeError

**Anything else we need to know?**:
Fix: Change the line to:
weights[key] = val.detach().cpu().numpy()
